dnf-behave-tests/features/steps/repo.py

```diff
@@ -107,6 +107,13 @@ def step_impl(context, rtype, repo):
     context.execute_steps('Given I use the repository "{}"'.format(repo_id))
 
 
+@behave.step("the server starts responding with HTTP status code {code}")
+def step_server_down(context, code):
+    path = context.dnf.repos_location
+    response = context.httpd.send_request(path, "STATUS %s" % code)
```

Comment:
Hey, this is a funky way to do this :slightly_smiling_face: is it not possible to simply set the configuration to the server directly?

If not, could you use HTTP arguments (the ones after a `?`) instead of putting it in the path?

Reply:
I'd be glad to learn how to do it (setting the configuration directly) :)  The server is a subprocess waiting on client connections and parsing their requests. I don't a simpler way to achieve this, to be honest :)

I do agree it's a bit funky, but to me it really seems to be the easiest solution.

One alternative would be to configure the server to respond with errors before actually spawning it, but then you would need to somehow pass arguments via the `httpd` fixture decorator and I'm not sure how to do that. I settled with this "dynamic" way since it might also be useful to "break" the server in a step dynamically.

Reply:
Also, what do you mean by using HTTP arguments? What's the server considers a "path" here is what goes after `STATUS`.

Reply:
OK, now that I talked with Lukas on IRC, it seems that it really is possible to do this cleanly; I'll just pass the configuration changes via shared memory and adjust the request handler to read from that memory :) Looks easy, I'll give it a try.